server/interaction/npc.go

```diff
@@ -0,0 +1,120 @@
+package interaction
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/joshprzybyszewski/cribbage/logic/scorer"
+	"github.com/joshprzybyszewski/cribbage/model"
+	"github.com/joshprzybyszewski/cribbage/utils/rand"
+)
+
+const (
+	Dumb   model.PlayerID = `DumbNPC`
+	Simple model.PlayerID = `SimpleNPC`
+	Calc   model.PlayerID = `CalculatedNPC`
+)
+
+var npcs = map[model.PlayerID]npcLogic{
+	Dumb:   &dumbNPCLogic{},
+	Simple: &simpleNPCLogic{},
+	Calc:   &calcNPCLogic{},
+}
+
+var _ Player = (*NPCPlayer)(nil)
+
+type NPCPlayer struct {
+	HandleActionCallback func(ctx context.Context, a model.PlayerAction) error
+	logic                npcLogic
+	id                   model.PlayerID
+}
+
+// NewNPCPlayer creates a new NPC with specified type
+func NewNPCPlayer(pID model.PlayerID, cb func(ctx context.Context, a model.PlayerAction) error) (Player, error) {
+	l, ok := npcs[pID]
+	if !ok {
+		return &NPCPlayer{}, errors.New(`not a valid npc mode`)
+	}
+	return &NPCPlayer{
+		logic:                l,
+		id:                   pID,
+		HandleActionCallback: cb,
+	}, nil
+}
+
+func (npc *NPCPlayer) ID() model.PlayerID {
+	return npc.id
+}
+
+func (npc *NPCPlayer) NotifyBlocking(b model.Blocker, g model.Game, s string) error {
+	a, err := npc.buildAction(b, g)
+	if err != nil {
+		return err
+	}
+	// TODO find a better way to solve this problem...
+	go func() {
+		time.Sleep(time.Second * 1)
```

Comment:
I think just having it in a go func() should be enough -- we shouldn't need to wait... :notsureif:

Reply:
erm for now I took it down to 20ms at least 🤷‍♂